accounts-db/store-histogram/src/main.rs

```diff
@@ -31,19 +31,31 @@ fn get_stars(x: usize, max: usize, width: usize) -> String {
     s
 }
 
-fn calc(info: &[(usize, usize)], bin_widths: Vec<usize>) {
+fn is_parsable<T>(string: String) -> Result<(), String>
+where
+    T: FromStr,
+    T::Err: Display,
+{
+    string
+        .parse::<T>()
+        .map(|_| ())
+        .map_err(|err| format!("error parsing '{string}': {err}"))
+}
+
+fn calc(info: &[(usize, usize)], bin_widths: Vec<usize>, offset: i64) {
     let mut info = info.to_owned();
     info.sort();
     let min = info.first().unwrap().0;
     let max_inclusive = info.last().unwrap().0;
+    let outside_slot = 432_000 - offset as usize;
     eprintln!("storages: {}", info.len());
     eprintln!("lowest slot: {min}");
     eprintln!("highest slot: {max_inclusive}");
     eprintln!("slot range: {}", max_inclusive - min + 1);
     eprintln!(
         "outside of epoch: {}",
         info.iter()
-            .filter(|x| x.0 < max_inclusive - 432_000)
+            .filter(|x| x.0 < max_inclusive - outside_slot)
```

Comment:
it is no longer an "epoch". 
Maybe change it to something like "ancient boundary"?

Reply:
changed.

Reply:
I think this print says "how many slots are there beyond ancient boundary"? 
```suggestion

 eprintln!(
        "ancient boundary: {}, num_of_slots_beyond_ancient_bondary: {}", outside_slot,
        info.iter()
            .filter(|x| x.0 < max_inclusive - outside_slot).count()
            ):
```

Reply:
done.